be/src/exec/es_http_scanner.cpp

```diff
@@ -0,0 +1,131 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exec/es_http_scanner.h"
+
+#include <sstream>
+#include <iostream>
+
+#include "runtime/descriptors.h"
+#include "runtime/exec_env.h"
+#include "runtime/mem_tracker.h"
+#include "runtime/raw_value.h"
+#include "runtime/runtime_state.h"
+#include "runtime/tuple.h"
+#include "exprs/expr.h"
+
+namespace doris {
+
+EsHttpScanner::EsHttpScanner(
+            RuntimeState* state,
+            RuntimeProfile* profile,
+            TupleId tuple_id,
+            const std::map<std::string, std::string>& properties,
+            const std::vector<ExprContext*>& conjunct_ctxs,
+            EsScanCounter* counter) :
+        _state(state),
+        _profile(profile),
+        _tuple_id(tuple_id),
+        _properties(properties),
+        _conjunct_ctxs(conjunct_ctxs),
+        _next_range(0),
+        _line_eof(false),
+        _batch_eof(false),
+#if BE_TEST
+        _mem_tracker(new MemTracker()),
+        _mem_pool(_mem_tracker.get()),
+#else 
+        _mem_tracker(new MemTracker(-1, "EsHttp Scanner", state->instance_mem_tracker())),
+        _mem_pool(_state->instance_mem_tracker()),
+#endif
+        _tuple_desc(nullptr),
+        _counter(counter),
+        _es_reader(nullptr),
+        _es_scroll_parser(nullptr),
+        _rows_read_counter(nullptr),
+        _read_timer(nullptr),
+        _materialize_timer(nullptr) {
+}
+
+EsHttpScanner::~EsHttpScanner() {
+    close();
```

Comment:
you should delete _es_scroll_parser here, otherwise there is a memory leak

Reply:
Change to use unique_ptr to avoid memory leak